src/logic/wallets/getWeb3.ts

```diff
@@ -64,7 +64,7 @@ export const getInfuraUrl = () => {
 export const web3ReadOnly =
   process.env.NODE_ENV !== 'test'
     ? new Web3(new Web3.providers.HttpProvider(getInfuraUrl()))
-    : new Web3((window as any).web3.currentProvider)
+    : new Web3((window as any).web3?.currentProvider || 'ws://localhost:8545')
```

Comment:
can we remove `any` here? This is how we should do it https://stackoverflow.com/questions/12709074/how-do-you-explicitly-set-a-new-property-on-window-in-typescript